state-chain/chains/src/lib.rs

```diff
@@ -536,3 +536,11 @@ impl RetryPolicy for DefaultRetryPolicy {
 		Some(10u32)
 	}
 }
+
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode, TypeInfo, MaxEncodedLen)]
+pub enum SwapType {
+	Swap(ForeignChainAddress),
+	CcmPrincipal(SwapId),
+	CcmGas(SwapId),
+	NetworkFee,
+}
```

Comment:
I don' think this should go in the Chains crate. 

Either we consider it part of the SwapQueueApi interface and define it there. Or we consider it a primitive and add it to primitives. But it has nothing to do with chains.

Reply:
was not sure where to put it. It can't go in primitives because it uses `ForeignChainAddress` from chains. So ill put it next to the `SwapQueueApi` for now.